### DISCUSSION TITLE
Which FluentUI Version To Use

### DISCUSSION THREAD

**Discussion by grumpykiwi** [microsoft/fluentui-blazor#1962] [General]
Hi

I noticed while stepping through the new Teams Tab app wizard that it still uses the old Fast (3.x) version of the FluentUI toolkit.

Is there a technical reason why I should not attempt to upgrade my little internal Teams Tab app to use the 4.x version of the toolkit ?

I use 4.x for everything else and am in the process of moving some older projects from other component toolkits to FluentUI. I like the consistency of the controls and the look and feel. Lets face it, Material is downright ugly. And the prices some of the vendors choose is highway robbery.

Anyway, just a question and an almost rant :-)

Cheers

Mark

**Comment by vnbaaij** [comment]
I think it should work with v4 without issues. They are probably still on v3 because of NET 6 support l.

**Comment by grumpykiwi** [comment]
Sweet. Makes sense. I’ll give it a spin here in a few. Cheers


Mark Nash

IT Director

(817) 465-9277  |  Arlington, Tx

Winner of 99 IABC Gold Quills and the 2023 Large Agency of the Year

[cid:linkedin_4c6f3bdf-dd4e-4f46-94f0-1885fa83094c.png]<https://www.linkedin.com/company/partnercomm>    [cid:instagram_bebedbe2-cb80-462a-af3c-7ae9caf0718c.png] <https://www.instagram.com/partnercomm>     [cid:facebook_a2d99c62-cb5a-42d3-b0e4-d08e71427149.png] <https://www.facebook.com/PartnerCommInc/>     [cid:twitter_53725725-ce5d-4ef3-bd47-bfd64fe511d7.png] <https://twitter.com/partnercomm>     [cid:vimeo_f71007e4-e318-469a-b82c-5751fc4fbe28.png] <https://vimeo.com/sweet250>                  [cid:web_a51bd891-a842-461a-b706-81937a9daae4.png]  www.partnercomm.net<https://www.partnercomm.net>
        [cid:logo_pcomm_7db038d0-366a-4820-b9c5-f8445cdf05ea.png]

________________________________

**Comment by vnbaaij** [comment]
Ok, let me know if you run into anything!

**Comment by grumpykiwi** [comment]
Upgraded painlessly overall. Replaced nuget packages and fixed a couple of usings. GenerateHeaderOption.Sticky is not working in my datagrid with virtualize=true and itemsize=60 set. Just checked production (v 3.62) and it is not working there either, so must be something off with my code. Looks fine to me. I have been fiddling with the same code in various versions and formats for darn near 3 years now, so won’t be surprised if the issue is staring back at me 😊

There are some things in program.cs that the Teams Toolkit puts in that look distinctly Net 7 or earlier. A bit weary of replacing them with .Net 8 things. No idea how the new rendering model stuff would work in Teams. That might be something for Jon Miller & Steve Sanderson to hash out 😊


      <FluentStack Orientation="Orientation.Horizontal" VerticalAlignment="VerticalAlignment.Center">
            <text style="font-size: 12pt; font-weight: 500;">Search: </text>
            <FluentSearch @***@***.***" @***@***.***" @bind-Value:after="HandleSearch" Placeholder="Search for a user" />
      </FluentStack>
      <br />
      <FluentDataGrid Items="GridData" TGridItem="GroupUser" ItemSize="60" Virtualize=true GridTemplateColumns="0.5fr 1.5fr 1.25fr 2.5fr 2fr 1.5fr 2fr 1.5fr"
            GenerateHeader="GenerateHeaderOption.Sticky" Style="background: white; width: 100%;" >
            <TemplateColumn Title="#" Align="Align.Center">
                <img class="statusImage" ***@***.***" ***@***.***" onclick="@(() => ShowModal(context.ID))" />
            </TemplateColumn>
            <TemplateColumn Title="Name">
                <a ***@***.***" target="_blank" style="text-decoration: none;">
                    <span ***@***.*** text-decoration: ***@***.***</span>
                    <i class="fa-sharp fa-light fa-comments" ***@***.*** vertical-align:bottom;"></i>
                </a>
            </TemplateColumn>
            <TemplateColumn Title="Status">
                <span ***@***.******@***.***</span>
            </TemplateColumn>
                  <TemplateColumn Title="Message">
                        @if (context.StatusMessageExpiration < DateTime.Now && !string.IsNullOrEmpty(context.StatusMessage))
                        {
                              var message = StringToHtml(context.StatusMessage.Replace("&nbsp;", "").Replace("<br />", "").Replace("<br>", ""));

                              <text>
                                    <span style="text-wrap: pretty; color: ***@***.***</span>
                              </text>
                        }
                        else
                        {
                              <text>&nbsp;</text>
                        }
                  </TemplateColumn>
                  <TemplateColumn Title="Email">
                        <a ***@***.***" >
                              <span style="text-decoration: underline; color: ***@***.***</span>
                              <i class="fa-sharp fa-light fa-envelope" style="vertical-align: bottom; color: var(--primary-medium);"></i>
                              </a>
                  </TemplateColumn>
                  <TemplateColumn Title="Phone">
                        @if (context.MobilePhone.Length > 0)
                        {
                              <text>
                                    <span nowrap="nowrap" style="color: var(--foreColor);">
                                          <b>(C)</b> @context.MobilePhone<br />
                                    </span>
                              </text>
                        }
                        @if (context.BusinessPhone.Length > 0 &&
                        string.Compare(context.BusinessPhone, context.MobilePhone, StringComparison.InvariantCulture) != 0)
                        {
                              <text>
                                    <span nowrap="nowrap" style="color: var(--foreColor);">
                                          <b>(O)</b> @context.BusinessPhone
                                    </span>
                              </text>
                        }
                  </TemplateColumn>
            <TemplateColumn Title="Title">
                <span style="word-wrap: normal; ***@***.***(context.JobTitle)</span>
            </TemplateColumn>
                  <TemplateColumn Title="Location">
                        @if (context.City.Length > 0 && context.State.Length > 0)
                        {
                              <span style="color: ***@***.***, @context.State</span>
                        }
                        else
                        {
                              if (!string.IsNullOrEmpty(context.OfficeLocation))
                              {
                                    <span style="color: var(--foreColor); text-wrap: pretty;" >@context.OfficeLocation</span>
                              }
                        }
                  </TemplateColumn>
        </FluentDataGrid>




Mark Nash

IT Director

(817) 465-9277  |  Arlington, Tx

Winner of 99 IABC Gold Quills and the 2023 Large Agency of the Year

[cid:linkedin_4c6f3bdf-dd4e-4f46-94f0-1885fa83094c.png]<https://www.linkedin.com/company/partnercomm>    [cid:instagram_bebedbe2-cb80-462a-af3c-7ae9caf0718c.png] <https://www.instagram.com/partnercomm>     [cid:facebook_a2d99c62-cb5a-42d3-b0e4-d08e71427149.png] <https://www.facebook.com/PartnerCommInc/>     [cid:twitter_53725725-ce5d-4ef3-bd47-bfd64fe511d7.png] <https://twitter.com/partnercomm>     [cid:vimeo_f71007e4-e318-469a-b82c-5751fc4fbe28.png] <https://vimeo.com/sweet250>                  [cid:web_a51bd891-a842-461a-b706-81937a9daae4.png]  www.partnercomm.net<https://www.partnercomm.net>
        [cid:logo_pcomm_7db038d0-366a-4820-b9c5-f8445cdf05ea.png]

**Comment by grumpykiwi** [comment]
Update.


 1.  Sticky header on data grid is working now. Not sure how now.
 2.  Changed program.cs to use the new Net 8 middleware calls such as builder.Services.AddRazorComponents().AddInteractiveServerComponents();
 3.  Changed project structure to Net 8 layout. All pages in components folder now. _hosts has been migrated to app.razor. Routes.razor added. MainLayout.razor created under \Components\Layout

All working except body font. Tried to set it to Montserrat (company style guide defined) using <BodyFont> method that Vincent clued me in on, but it is still defaulting to Segoe. Think it is coming from something in Teams. Have yet to find a way to override it.

I am pretty happy with the progress. Will take another look at the font think later or tomorrow, depending on the ominous looking weather heading my way here in North Texas.

Cheers



Mark Nash

IT Director

(817) 465-9277  |  Arlington, Tx

Winner of 99 IABC Gold Quills and the 2023 Large Agency of the Year

[cid:linkedin_4c6f3bdf-dd4e-4f46-94f0-1885fa83094c.png]<https://www.linkedin.com/company/partnercomm>    [cid:instagram_bebedbe2-cb80-462a-af3c-7ae9caf0718c.png] <https://www.instagram.com/partnercomm>     [cid:facebook_a2d99c62-cb5a-42d3-b0e4-d08e71427149.png] <https://www.facebook.com/PartnerCommInc/>     [cid:twitter_53725725-ce5d-4ef3-bd47-bfd64fe511d7.png] <https://twitter.com/partnercomm>     [cid:vimeo_f71007e4-e318-469a-b82c-5751fc4fbe28.png] <https://vimeo.com/sweet250>                  [cid:web_a51bd891-a842-461a-b706-81937a9daae4.png]  www.partnercomm.net<https://www.partnercomm.net>
        [cid:logo_pcomm_7db038d0-366a-4820-b9c5-f8445cdf05ea.png]
